app/src/main/java/de/westnordost/streetcomplete/overlays/surface/PathSurfaceOverlay.kt

```diff
@@ -0,0 +1,146 @@
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.WHEELCHAIR
+import de.westnordost.streetcomplete.osm.ALL_PATHS
+import de.westnordost.streetcomplete.osm.ALL_ROADS
+import de.westnordost.streetcomplete.osm.isPrivateOnFoot
+import de.westnordost.streetcomplete.osm.sidewalk.Sidewalk
+import de.westnordost.streetcomplete.osm.sidewalk.createSidewalkSides
+import de.westnordost.streetcomplete.osm.sidewalk_surface.createSidewalkSurface
+import de.westnordost.streetcomplete.osm.surface.SurfaceAndNote
+import de.westnordost.streetcomplete.osm.surface.UNDERSPECIFED_SURFACES
+import de.westnordost.streetcomplete.osm.surface.UnknownSurface
+import de.westnordost.streetcomplete.osm.surface.createSurfaceStatus
+import de.westnordost.streetcomplete.overlays.Color
+import de.westnordost.streetcomplete.overlays.Overlay
+import de.westnordost.streetcomplete.overlays.PolygonStyle
+import de.westnordost.streetcomplete.overlays.PolylineStyle
+import de.westnordost.streetcomplete.overlays.StrokeStyle
+import de.westnordost.streetcomplete.overlays.Style
+import de.westnordost.streetcomplete.quests.surface.AddCyclewayPartSurface
+import de.westnordost.streetcomplete.quests.surface.AddFootwayPartSurface
+import de.westnordost.streetcomplete.quests.surface.AddPathSurface
+import de.westnordost.streetcomplete.quests.surface.AddSidewalkSurface
+
+class PathSurfaceOverlay : Overlay {
+
+    override val title = R.string.overlay_path_surface
+    override val icon = R.drawable.ic_quest_way_surface
+    override val changesetComment = "Specify path surfaces"
+    override val wikiLink: String = "Key:surface"
+    override val achievements = listOf(PEDESTRIAN, WHEELCHAIR, BICYCLIST, OUTDOORS)
+    override val hidesQuestTypes = setOf(
+        AddPathSurface::class.simpleName!!,
+        AddFootwayPartSurface::class.simpleName!!,
+        AddCyclewayPartSurface::class.simpleName!!,
+        AddSidewalkSurface::class.simpleName!!,
+    )
+
+    override fun getStyledElements(mapData: MapDataWithGeometry): Sequence<Pair<Element, Style>> =
+        mapData.filter( """
+               ways, relations with
+                   highway ~ ${(ALL_PATHS).joinToString("|")}
+                   and (segregated = yes or (!cycleway:surface and !footway:surface and !cycleway:surface:note and !footway:surface:note))
+                   and (!surface:note or (surface or cycleway:surface or footway:surface or segregated = yes))
+                   and (!cycleway:surface:note or cycleway:surface)
+                   and (!footway:surface:note or footway:surface)
+                   and !sidewalk and !sidewalk:left and !sidewalk:right and !sidewalk:both
+                   and !sidewalk:both:surface and !sidewalk:right:surface and !sidewalk:left:surface and !sidewalk:surface
+                   and !sidewalk:both:surface:note and !sidewalk:right:surface:note and !sidewalk:left:surface:note
```

Comment:
> \50) the surface of an unsegregated path that does have (wrongly?) a specific surface for the cycleway or footway part set. Why not just let the user set the surface in that case and not touch footway:surface etc.?

This complicates deciding how to display it and handle - and anyway there is isSegregated quest that should trigger in such cases and provide us `segregated` tag

> 51-53) surfaces that have a note but not the surface itself set (yet). The form could just display this and not allow saving it without changes because it is incomplete isFormComplete() returns false.

It used to be there so I could ignore that part of complexity and it was extremely really rare anyway. But now I think it is supported anyway so it can return

> 54-56) why not let user specify the main surface of a path with a sidewalk? (E.g. cycleways with sidewalks). Probably it would be best to treat cycleways with sidewalks like roads in respect to the RoadSurfaceOverlay and PathSurfaceOverlay quests - enable editing the surface of such cycleways in the RoadSurfaceOverlay overlay and enable editing the sidewalk surface of such cycleways in the PathSurfaceOverlay overlay.

Again, to allow ignoring this and to pretend that it does not exists. Not entirely sure what is the best way to support it, half-seriously my plan was to wait for someone to open an issue asking for support of such way and ask them (not sure is it OK to treat it as a road with sidewalks)

I also have idea for future to potentially merge path and road layer and have GUI to allow changing both sidewalk surface and main road surface

Reply:
(that is explanation how it ended in this state, not sure yet myself which ones should be eliminated)

Reply:
> 51-53) surfaces that have a note but not the surface itself set (yet). The form could just display this and not allow saving it without changes because it is incomplete isFormComplete() returns false.

this was tested and removed as no longer needed